canonical-web-and-design/maas-ui
# feat(ui): handle redirects for the different KVM details pages

## Done

- Added a hook that determines the cluster of a host, if it exists.
- Added a hook that handles redirects to and from LXD single host, LXD cluster host and virsh host depending what the host is.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Try navigating to the LXD single host URL of a host that's actually in a cluster, and check that you get redirected to the cluster version. e.g. using bolla, navigate to `/MAAS/r/kvm/lxd/484` and check that you get redirected to `/MAAS/r/kvm/lxd/cluster/22/vms/484`
- Try navigating to the LXD single host URL for a virsh host, e.g. using bolla navigate to `/MAAS/r/kvm/lxd/455` and check you get redirected to `/MAAS/r/kvm/virsh/455`
- Try navigating to the virsh host URL for a LXD single host, e.g. using bolla navigate to `/MAAS/r/kvm/virsh/336` and check you get redirected to `/MAAS/r/kvm/lxd/336`

## Fixes

Fixes canonical-web-and-design/app-squad#341
Fixes canonical-web-and-design/app-squad#401
